### ISSUE TITLE
Abrupt "peer connection reset" and timeout  from the ESX service

### ISSUE THREAD

**Issue by msterin** [vmware/docker-volume-vsphere#497]
https://ci.vmware.run/vmware/docker-volume-vsphere/730 "DOCKER RM TEST" failed due to the reason below. We the same failures at different times in different tests.  We can hide it with retries (and in fact should do that to unblock the CI) but we need to understand the actual bug.
 
```
	refcnt_test.go:76: 2016-06-29T21:08:24Z " docker volume rm refCountTestVol" failed but expected to succeed
	refcnt_test.go:35: exit status 1
```
Looking in the client logs: 
```
2016-06-29 21:08:24.676700775 +0000 UTC [WARNING] 'remove' failed: connection timed out (errno=110).
2016-06-29 21:08:24.676747134 +0000 UTC [ERROR] Failed to remove volume name=refCountTestVol error="'remove' failed: connection timed out (errno=110)." 
2016-06-29 21:08:24.844217873 +0000 UTC [INFO] Removing volume name=refCountTestVol 
2016-06-29 21:08:24.844419422 +0000 UTC [WARNING] 'remove' failed: connection reset by peer (errno=104). Hit communication issue with ESX (vmci or ESX service)
```
Server did not see an issue: 
```
06/29/16 21:08:06 237715 [photon.rc.vmfs] [INFO   ] Disk detached /vmfs/volumes/datastore1/dockvols/refCountTestVol.vmdk
06/29/16 21:08:06 237715 [photon.rc.vmfs] [INFO   ] executeRequest 'detach' completed with ret=None
06/29/16 21:08:06 237715 [photon.rc.vmfs] [INFO   ] executeRequest 'get' completed with ret=None
06/29/16 21:08:08 237715 [photon.rc.vmfs] [INFO   ] executeRequest 'get' completed with ret=None
```
Sometimes we see similar failure but with "Peer connection reset" 

umbrella issue: #489

**Comment by pdhamdhere** [comment]
#499 introduced retries thus dropping Blocker tag

**Comment by pdhamdhere** [comment]
Fixed with #532
